### ISSUE TITLE
Adopt Jakarta EE 10

### ISSUE THREAD

**Issue by Emily-Jiang** [Azure-Samples/open-liberty-on-aks#43]
This application uses Java EE 8 and it is out of date. I would like this application to be updated to Jakarta EE 10 as @bbenz and I will need to demo this at DevNexus at the Jakarta EE track. I don't think using a Java EE 8 app is appropriate for a Jakarta EE talk. I am happy to do a PR with the namespace transform. Please let me know what you think.

**Comment by Emily-Jiang** [comment]
@edburns @majguo please let me know what you think

**Comment by edburns** [comment]
Iterating in #44 .

**Comment by edburns** [comment]
Hello @Emily-Jiang ,

I am working on your #43.

I am trying to configure it for use with Liberty 23.0.0.3. There is a problem in the `pom.xml/server.xml`. Even with a carefully vetted set of `<feature>` elements in the `<featureManager>` I still get this error at build time.

```
[ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.10:install-feature (package-server) on project javaee-cafe: Error installing features for server defaultServer: CWWKF1299E: The following features could not be obtained: jaxb-4.0 restfulws-3.0 enterprisebeanslite-4.0 cdi-4.0 persistence-3.0 faces-4.0 io.openliberty.emptyHandleListContext-1.0 com.ibm.websphere.appserver.ejbliteJNDI-1.0 com.ibm.websphere.appserver.javaeePlatform7.0-jndi1.0 io.openliberty.ejbliteJPA-2.0 io.openliberty.restfulWS3.0-xmlBinding3.0 com.ibm.websphere.appserver.ejbComponentMetadataDecorator-1.0 com.ibm.websphere.appserver.jakarta-transactionContext-1.0 com.ibm.websphere.appserver.classloaderContext-1.0. Ensure that the features are valid for Open Liberty. -> [Help 1]
```

Can you please take a look at the [pom.xml](https://github.com/azure-javaee/open-liberty-on-aks/blob/edburns-msft-ibm-524-update-to-ee-10/java-app/pom.xml) and [server.xml](https://github.com/azure-javaee/open-liberty-on-aks/blob/edburns-msft-ibm-524-update-to-ee-10/java-app/src/main/liberty/config/server.xml) and see if anything jumps out to you?

**Comment by edburns** [comment]
Problem https://github.com/Azure-Samples/open-liberty-on-aks/issues/43#issuecomment-1947605954 was fixed in c8d92a1 .

**Comment by edburns** [comment]
I need help getting past this error: `Could not find MessageBodyWriter for response object of type: java.util.Vector of media type: application/json`. Please feel free to push code to the ranch.

Deployment details [See attachment 20240216-ibm-569-deployment-details.md](https://dev.azure.com/was-azure/was-azure/_workitems/edit/569).

## Steps to reproduce
1. Clone this repo and checkout this branch: https://github.com/azure-javaee/open-liberty-on-aks/tree/edburns-msft-ibm-524-update-to-ee-10 
1. Set your JDK to be JDK 17.
1. Set your environment variables as in 20240216-ibm-569-deployment-details.md.
1. cd java-app
1. mvn clean install
1. mvn liberty:run
1. curl -H "Accept: application/xml" http://localhost:9080/rest/coffees 
   - Note that it provides a correct result:
      ```xml
      <?xml version="1.0" encoding="UTF-8" standalone="yes"?><collection><coffee><id>256</id><name>decaf</name><price>1.0</price></coffee></collection>
      ```
1. curl -H "Accept: application/json" http://localhost:9080/rest/coffees
   - Note that it provides an incorrect result:
   ```
   Could not find MessageBodyWriter for response object of type: java.util.Vector of media type: application/json
   ```

## Measures tried so far

- Verified that Cafe.getAllCoffees() has the same problem as with the simple cURL.
- Verified I can POST a coffee
   `curl -X POST -H "Content-Type: application/xml" -d @coffee.xml http://localhost:9080/rest/coffees`
   ```xml
   <coffee>
       <id>256</id>
       <name>decaf</name>
       <price>1.0</price>
   </coffee>
   ```
- Try to force it to have a provider for Json from the resteasy-json-binding-provider. Ran in to a bunch of classloading problems. I believe the provider is already there, but it's not being found for some reason. I backed out these changes.
